Sources/Atomics/AtomicStrongReference.swift

```diff
@@ -13,17 +13,59 @@
 import _AtomicsShims
 
 /// A class type that supports atomic strong references.
+///
+///     class MyObject: AtomicReference {}
+///
+///     let object = MyObject()
+///     let ref = ManagedAtomic<MyObject>(object)
+///
+///     ref.load(ordering: .relaxed) // Returns `object`.
+///
+/// The conforming class is allowed to be non-final, but `ManagedAtomic` and
+/// `UnsafeAtomic` do not support using a subclass as their generic argument --
+/// the type of an atomic reference must be precisely the same class that
+/// originally conformed to the protocol.
+///
+///
+///     class Derived: MyObject {}
+///
+///     let ref2: ManagedAtomic<Derived>
+///     // error: 'ManagedAtomic' requires the types 'Derived' and 'Base' be equivalent
+///
+/// Note that this limitation only affects the static type of the
+/// `ManagedAtomic`/`UnsafeAtomic` variables. Such references still fully
+/// support holding instances of subclasses of the conforming class. (Returned
+/// may be downcasted from the base type after an `is` check.)
+///
+///     let child = Derived()
+///     ref.store(child, ordering: .relaxed) // OK!
+///     let value = ref.load(ordering: .relaxed)
+///     // `value` is a variable of type `MyObject`, holding a `Derived` instance.
+///     print(value is Derived) // Prints "true"
+///
 public protocol AtomicReference: AnyObject, AtomicOptionalWrappable
 where
-  AtomicRepresentation == AtomicReferenceStorage<Self>,
-  AtomicOptionalRepresentation == AtomicOptionalReferenceStorage<Self>
+  AtomicRepresentation == AtomicReferenceStorage<_AtomicBase>,
+  AtomicOptionalRepresentation == AtomicOptionalReferenceStorage<_AtomicBase>
```

Comment:
@lorentey Wow, this is a cool pattern. I did not know you could use a recursive type constraint within the where clause in the protocol inheritance itself! 

Any documentation on how this was available?